src/core/IncomingRequestManager.cpp

```diff
@@ -192,9 +331,11 @@ void IncomingContactRequest::load()
 
 void IncomingContactRequest::save()
 {
-    SettingsObject settings(QStringLiteral("contactRequests.%1").arg(QLatin1String(m_hostname)));
+    SettingsObject settings(settingsKey());
 
+#ifndef PROTOCOL_NEW
     settings.write("remoteSecret", Base64Encode(remoteSecret()));
```

Comment:
The secret used to be required, and is no longer required; it's all based on keys now. So, no problems.